crates/matrix-sdk-crypto/src/types/one_time_keys.rs

```diff
@@ -0,0 +1,210 @@
+// Copyright 2021 The Ruma contributors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+//! Module containing customized types modeling Matrix one-time keys.
+//!
+//! These types were mostly taken from the Ruma project. The types differ in two
+//! important ways to the Ruma types of the same name:
+//!
+//! 1. They are using vodozemac types so we directly deserialize into a
+//!    vodozemac curve25519 key.
+//! 2. They support lossless serialization cycles.
```

Comment:
Define "lossless" 😉

I think it might be worth it to note "except for whitespace and field order".

Reply:
I guess I should mention lossless in a canonical JSON type of way, which avoids the white-space and field order issue.

Reply:
Did so https://github.com/matrix-org/matrix-rust-sdk/pull/548/commits/3024294120d06e3845cb74ae749e9d8869c6e340.